lumeer-storage/lumeer-storage-mongodb/src/main/java/io/lumeer/storage/mongodb/dao/system/MongoUserDao.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.lumeer.storage.mongodb.dao.system;
+
+import static io.lumeer.storage.mongodb.util.MongoFilters.idFilter;
+
+import io.lumeer.api.model.User;
+import io.lumeer.storage.api.dao.UserDao;
+import io.lumeer.storage.api.exception.StorageException;
+import io.lumeer.storage.mongodb.MongoUtils;
+import io.lumeer.storage.mongodb.codecs.UserCodec;
+import io.lumeer.storage.mongodb.model.MongoUser;
+
+import com.mongodb.MongoException;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.Indexes;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.result.UpdateResult;
+
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.RequestScoped;
+
+@RequestScoped
+public class MongoUserDao extends SystemScopedDao implements UserDao {
+
+   private static final String COLLECTION_NAME = "users";
+   private static final String ELEMENT_NAME = "group";
+
+   @PostConstruct
+   public void checkRepository() {
+      if (database.getCollection(COLLECTION_NAME) == null) {
+         createUsersRepository();
+      }
+   }
+
+   public void createUsersRepository() {
+      database.createCollection(databaseCollectionName());
+
+      MongoCollection<Document> userCollection = database.getCollection(databaseCollectionName());
+      userCollection.createIndex(Indexes.ascending(UserCodec.EMAIL), new IndexOptions().unique(true));
```

Comment:
I think we would also need an index on `keycloakId` field.

Reply:
Okay